File: frontend/src/types/reservation.ts
export interface Reservation {
  id: number;
  tableId: number;
  guestName: string;
  partySize: number;
  startsAt: string;   // ISO 8601
  endsAt: string;     // startsAt + 2.5h
  notes: string | null;
  createdAt: string;
}

export interface ReservationRequest {
  tableId: number;
  guestName: string;
  partySize: number;
  startsAt: string;
  notes?: string;
}

export interface ReservationFilter {
  date?: string;          // ISO date "2026-03-15"
  partySize?: number;
  areaId?: number;
}

File: backend/src/main/java/com/restorano/backend/reservation/models/Reservation.java
package com.restorano.backend.reservation.models;

import com.restorano.backend.layout.models.RestaurantTable;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.Instant;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(schema = "restorano", name = "reservation")
@Getter
@Setter
@NoArgsConstructor
public class Reservation {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "guest_name", nullable = false)
    private String guestName;

    @Column(name = "party_size", nullable = false)
    private int partySize;

    @Column(name = "starts_at", nullable = false)
    private Instant startsAt;

    @Column(name = "ends_at", nullable = false)
    private Instant endsAt;

    @Column(name = "notes")
    private String notes;

    @Column(name = "created_at", updatable = false)
    private Instant createdAt = Instant.now();

    @ManyToMany
    @JoinTable(
        schema = "restorano",
        name = "reservation_tables",
        joinColumns = @JoinColumn(name = "reservation_id"),
        inverseJoinColumns = @JoinColumn(name = "table_id")
    )
    private List<RestaurantTable> tables = new ArrayList<>();
}
